WordPress/Sniffs/PHP/IniSetSniff.php

```diff
@@ -0,0 +1,173 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @link    https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+namespace WordPressCS\WordPress\Sniffs\PHP;
+
+use WordPressCS\WordPress\AbstractFunctionParameterSniff;
+use \PHP_CodeSniffer\Util\Tokens;
+
+/**
+ * Checks use of ini_set function with a blacklist for errors.
+ *
+ * @package WPCS\WordPressCodingStandards
+ *
+ * @since   x.x.x
+ */
+class IniSetSniff extends AbstractFunctionParameterSniff {
+
+	/**
+	 * Array of functions that must be checked
+	 *
+	 * @since x.x.x
+	 *
+	 * @var array Multidimensional array with parameter details.
+	 *     $target_functions = array(
+	 *         (string) Function name.
+	 *     );
+	 */
+	protected $target_functions = array(
+		'ini_set' => array(),
+	);
+
+	/**
+	 * Array of options that are allowed to be manipulated
+	 *
+	 * @since x.x.x
+	 *
+	 * @var array Multidimensional array with parameter details.
+	 *     $whitelisted_options = array(
+	 *         (string) option name. = array(
+	 *             (string[]) 'valid_values' = array()
+	 *         )
+	 *     );
```

Comment:
> Just wondering: is there a particular reason you've chosen to make this a three, rather than a two dimensional array with the valid value directly in the `option_name` array?

The array key makes the intent explicit. Also is more flexible this way.